src/compiler/program.ts

```diff
@@ -3,7 +3,7 @@
 /// <reference path="core.ts" />
 
 namespace ts {
-    const ignoreDiagnosticCommentRegEx = /(^\s*$)|(^\s*\/\/\/?\s*(@ts-ignore)?)/;
+    const ignoreDiagnosticCommentRegEx = /(^\s*$)|(^\s*\/\/\/?\s*(@ts-ignore)?(?:\s+((?:(?:TS\d+),\s*)*(?:TS\d+)?))?)/;

```

Comment:
The `?` in `(?:TS\d+)?` is redundant (causes an extra backtrack into a 0-length match, which is already allowed by the `?` in the outer non-capturing match)

Reply:
agree, thanks,  I'll remove it later